### PR TITLE
Add ability to use custom po files inside plugins

### LINKED ISSUE #16923: Allow multiple .po files in plugin to be declared and loaded
### Description

Currently it is not possible to load multiple `.po` files inside **a plugin** because we only check for the plugin specific translation file via https://github.com/cakephp/cakephp/blob/4.x/src/I18n/MessagesFileLoader.php#L173-L179

So users can only have one  `my_plugin.po` and `my_plugin.mo` inside the `plugins/MyPlugin/resources/locales/<lang>/` folder but can't split that one up if it gets to big. 

This is however possible from a pure app perspective because the searchpath for the app locales folder is always added.
https://github.com/cakephp/cakephp/blob/4.x/src/I18n/MessagesFileLoader.php#L162-L170

## Working app example

* Add this to `config/bootstrap.php`
```
use Cake\I18n\MessagesFileLoader as Loader;
\Cake\I18n\I18n::setTranslator(
    'random',
    new Loader('random', 'de_DE', 'po'),
    'de_DE'
);
```
* add `<?= __d('random', 'Translate this text') ?>` to a template file located in the apps template folder
* extract the `.pot` files via this command
```
bin/cake i18n extract --paths $PWD/config/,$PWD/src/,$PWD/templates --overwrite --output $PWD/resources/locales/de_DE --extract-core yes
```
* This now generated a `resources/locales/de_DE/random.pot` which can be translated to a `random.po` and `random.mo`

Now the additional `random.po` file can be used to split up translations from the `default.po`

## Thoughts about a possible fix

It would be nice to have the same dot-notation like `PluginName.random` present for `__d()` as we have it in other places like `PluginName.MyController` or `PluginName.MyTable`

WIth that one can easily re-use translation files, no matter where they are placed or how they are structured.

### CakePHP Version

4.4

### PR DESCRIPTION
**Pull request by ajibarra**
This PR fixes #16923 

The approach I have followed:

1. Keep backward compatibility. You can still use __d(plugin_name, string) and it will load plugin_name.po from plugin as default, not needing '.' notation
2. Add ability to use __d(plugin_name.custom) to allow loading custom.po from plugin.
4. Allow override of plugin po file using one in app using:

```
\Cake\I18n\I18n::setTranslator(
    'plugin_name.custom',
    new Loader('custom', 'es_ES', 'po'),
    'es_ES'
);
```
5. Allow override of app po file using one in plugin using:

```
\Cake\I18n\I18n::setTranslator(
    'custom',
    new Loader('plugin_name.custom', 'es_ES', 'po'),
    'es_ES'
);
```

I have improved the current unit test adding a new assertion using '.' notation. Please let me know if you like this approach. I could include more unit tests for different scenarios.

### CODE REVIEW

**Comment by LordSimal** [comment]
This looks awesome! I love it 👍🏻

File: tests/TestCase/I18n/I18nTest.php [hunk 1]
```diff
@@ -167,6 +167,12 @@ public function testPluginMesagesLoad(): void
             'String 1 (from plugin three)',
             $translator->translate('String 1')
         );
+
+        $translator = I18n::getTranslator('company/test_plugin_three.custom');
```
Comment (markstory) [inline tests/TestCase/I18n/I18nTest.php:171]: It would be good to have an integration test with `__d()` covering the userland use case as well.

**Comment by ajibarra** [comment]
@markstory I have added more asserts to test __d function, please let me know if they are enough.

@markstory @dereuromark I am not sure why the pipeline is failing now. This task `testsuite (7.4, mysql, prefer-lowest)` is throwing an error but it does not make sense to me. If you have any insight would be great.

**Comment by dereuromark** [comment]
The fail looks like sth already an issue on upstream (4.next). So not related to your changes.

**Pull request merged**